File: CXSJ3/Translate.py
from transformers import AutoTokenizer, AutoModelForSeq2SeqLM

def load_tokenizer(path):
    return AutoTokenizer.from_pretrained(path)

def load_model(path):
    return AutoModelForSeq2SeqLM.from_pretrained(path)

def translate(tokenizer, model, text):
    translated = model.generate(**tokenizer.prepare_seq2seq_batch([text], return_tensors="pt"))
    return [tokenizer.decode(t, skip_special_tokens=True) for t in translated]

File: Application/Photo/Translate.swift
//
//  Translate.swift
//  Photo
//
//  Created by 张力允 on 2020/12/16.
//

import Foundation
import PythonKit

class Translate
{
    private let tokenizer: PythonObject?
    private let model: PythonObject?
    private let sys: PythonObject
    private let translateFile: PythonObject
    private let path = "/Users/zhangliyun/Developer/CXSJ3/opus-mt-en-zh"
    
    init() throws
    {
        sys = Python.import("sys")
        sys.path.append("/Users/zhangliyun/Developer/CXSJ3")
        translateFile = Python.import("Translate")
        
        let os = Python.import("os")
        if let flag = Bool(os.path.exists(path)), flag
        {
            tokenizer = translateFile.load_tokenizer(path)
            model = translateFile.load_model(path)
        }
        else
        {
            throw myErrors.fileNotExists("File Doesn't Exist", "Translation model dosen't exist, please contact the developer. ")
        }
    }
    
    func translate(_ text: String) -> String?
    {
        if let tokenizer = self.tokenizer, let model = self.model
        {
            return Array(translateFile.translate(tokenizer, model, text))?[0]
        }
        return nil
    }
}
